lib/timers.js

```diff
@@ -596,14 +596,13 @@ function processImmediate() {
   const queue = outstandingQueue.head !== null ?
     outstandingQueue : immediateQueue;
   var immediate = queue.head;
-  const tail = queue.tail;
 
   // Clear the linked list early in case new `setImmediate()` calls occur while
   // immediate callbacks are executed
-  queue.head = queue.tail = null;
-
-  let count = 0;
-  let refCount = 0;
+  if (queue !== outstandingQueue) {
+    queue.head = queue.tail = null;
+    immediateInfo[kHasOutstanding] = 1;
```

Comment:
Maybe we could leave a comment here about how this `= null` thing works? it's... pretty difficult to read/understand.

Reply:
we should disallow that in the lint rules. please split it into two.